ast/util.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast
+
+import (
+	"strings"
+)
+
+// TextElement is an element of a SQLSentence.
+type TextElement struct {
+	// Next is the next element in a SQLSentence.
+	Next *TextElement
+
+	// Next is the previous element in a SQLSentence.
+	Prev *TextElement
+
+	// The text stored with this element.
+	Text string
+}
+
+type sqlSentenceStatus int8
+
+const (
+	sqlSentenceNormal sqlSentenceStatus = iota
+	sqlSentenceNearBrackets
+	sqlSentenceInBrackets
+)
+
+// SQLSentence express a sql text.
+// In order to avoid copying when string splicing occurs,
+// SQLSentence uses a linked list structure to store strings.
```

Comment:
When we will insert a Sentence in the **middle** of another Sentence, instead of the end? Appending Sentences can be easier done using a `strings.Builder` directly.